extensions/SharkPool/Spotify.js

```diff
@@ -0,0 +1,506 @@
+          response = lastFetchedSong[1];
+        } else {
+          if (!await Scratch.canFetch(`${proxy}https://open.spotify.com/track/${args.URL}`)) return "";
+          // eslint-disable-next-line
+          response = await fetch(`${proxy}https://open.spotify.com/track/${args.URL}`);
+          if (!response.ok) return "Error: 404";
+          response = await response.text();
+          lastFetchedSong = [args.URL, response];
+        }
+        let regex;
+        let match;
+        switch (args.THING) {
+          case "artist":
+            regex = /- song and lyrics by ([^|]+) \| Spotify<\/title>/;
+            match = response.match(regex);
+            match = this.replaceSpecial(match);
+            return match && match[1] ? match[1] : "";
+          case "artist ID":
+            regex = /<meta name="music:musician" content="https:\/\/open.spotify.com\/artist\/([^"]+)"\/>/;
+            match = response.match(regex);
+            return match && match[1] ? match[1] : "";
+          case "cover":
+            regex = /<meta name="twitter:image" content="([^"]+)"\/>/;
+            match = response.match(regex);
+            return match && match[1] ? match[1] : "";
+          case "release date":
+            regex = /<meta name="music:release_date" content="([^"]+)"\/>/;
+            match = response.match(regex);
+            return match && match[1] ? match[1] : "";
+          case "length":
+            regex = /<meta name="music:duration" content="([^"]+)"\/>/;
+            match = response.match(regex);
+            return match && match[1] ? match[1] + " seconds" : "";
+          case "listens":
+            regex = /"type">([^<]+)<\/span><\/a><\/div><\/div><button/;
+            match = response.match(regex);
+            return match && match[1] ? match[1] : "";
+          default:
+            regex = /<meta property="og:title" content="([^"]+)"\/>/;
+            match = response.match(regex);
+            match = this.replaceSpecial(match);
+            return match && match[1] ? match[1] : "";
+        }
+      } catch (error) {
+        return `Error: ${error}`;
+      }
+    }
+
+    async getArtistAtt(args) {
+      try {
+        let response;
+        if (lastFetchedArtist[0] === args.URL) {
+          response = lastFetchedArtist[1];
+        } else {
+          if (!await Scratch.canFetch(`${proxy}https://open.spotify.com/artist/${args.URL}`)) return "";
+          // eslint-disable-next-line
+          response = await fetch(`${proxy}https://open.spotify.com/artist/${args.URL}`);
+          if (!response.ok) return "Error: 404";
+          response = await response.text();
+          lastFetchedArtist = [args.URL, response];
+        }
+        let regex;
+        let match;
+        switch (args.THING) {
+          case "description":
+            regex = /<span[^>]*class="Type__TypeElement-sc-goli3j-0 gLgnHU G_f5DJd2sgHWeto5cwbi"[^>]*data-encore-id="type">([\s\S]*?)<\/span><\/div><\/div><\/div><\/div>/g;
```

Comment:
seems awfully fragile

Reply:
Unfortunately it has to be that way since Spotify is stupid and named all their classes practically the same